package/MDAnalysis/lib/formats/libdcd.pyx

```diff
@@ -0,0 +1,702 @@
+        # This function assumes that the dcd header was already read and
+        # self.ndims is set. It will only work when called here !!!
+        self.n_frames = self._estimate_n_frames()
+        self.b_read_header = True
+
+        # make sure fixed atoms have been read
+        try:
+            self.read()
+            self.seek(0)
+        except IOError:
+            if self.n_frames != 0:
+                raise IOError("DCD is corrupted")
+
+        if sys.version_info[0] < 3:
+            py_remarks = unicode(py_remarks, 'ascii', "ignore")
+            py_remarks = str(py_remarks.encode('ascii', 'ignore'))
+        else:
+            if isinstance(py_remarks, bytes):
+                py_remarks = py_remarks.decode('ascii', 'ignore')
+
+        py_remarks = "".join(s for s in py_remarks if s in string.printable)
+
+        self.remarks = py_remarks
+
+    cdef int _estimate_n_frames(self):
+        """ Only call this function in _read_header!!!
+        """
+        extrablocksize = 48 + 8 if self.charmm & DCD_HAS_EXTRA_BLOCK else 0
+        self._firstframesize = (self.natoms + 2) * self.ndims * sizeof(float) + extrablocksize
+        self._framesize = ((self.natoms - self.nfixed + 2) * self.ndims * sizeof(float) +
+                          extrablocksize)
+        filesize = path.getsize(self.fname)
+        # It's safe to use ftell, even though ftell returns a long, because the
+        # header size is < 4GB.
+        self._header_size = fio_ftell(self.fp)
+        nframessize = filesize - self._header_size - self._firstframesize
+        return nframessize / self._framesize + 1
+
+    @property
+    def is_periodic(self):
+        """
+        Returns
+        -------
+        bool if periodic unitcell is available
+        """
+        return bool((self.charmm & DCD_IS_CHARMM) and
+                    (self.charmm & DCD_HAS_EXTRA_BLOCK))
+
+    def seek(self, frame):
+        """Seek to Frame.
+
+        Parameters
+        ----------
+        frame : int
+            seek the file to given frame
+
+        Raises
+        ------
+        IOError
+            If you seek for more frames than are available or if the
+            seek fails (the low-level system error is reported).
+        """
+        if frame >= self.n_frames:
+            raise EOFError('Trying to seek over max number of frames')
+        self.reached_eof = False
+
+        cdef fio_size_t offset
+        if frame == 0:
+            offset = self._header_size
+        else:
+            offset = self._header_size + self._firstframesize + self._framesize * (frame - 1)
+
+        ok = fio_fseek(self.fp, offset, _whence_vals["FIO_SEEK_SET"])
+        if ok != 0:
+            raise IOError("DCD seek failed with system errno={}".format(DCD_ERRORS[ok]))
+        self.current_frame = frame
+
+    @property
+    def header(self):
+        """
+        Returns
+        -------
+        dict of header values needed to write new dcd.
```

Comment:
This doesn't format very great. It's OK. I don't know how to get super pretty formatting here though.